modules/layers/src/text-layer/text-layer.js

```diff
@@ -119,17 +119,21 @@ export default class TextLayer extends CompositeLayer {
   }
 
   getPickingInfo({info}) {
+    // because `TextLayer` assign the same pickingInfoIndex for one text label,
+    // here info.index refers the index of text label in props.data
     return Object.assign(info, {
       // override object with original data
-      object: info.object && info.object.object
+      object: info.index > -1 && this.props.data[info.index]
```

Comment:
This syntax is fine when used with an `if` statement. However, the pick info object is a user-facing API. Doing a shorthand will set `object` to `false` when nothing is picked.

```js
info.index >= 0 ? this.props.data[info.index] : null
```